### PR TITLE
add compileArticle hook to api documentation

### LINKED ISSUE #281: [3.5] add compileArticle hook to api documentation

### PR DESCRIPTION
**Pull request by bezin**
Fixes #281

### CODE REVIEW

File: api/hooks/compileArticle.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+The `compileArticle` hook is triggered after the article module has been compiled. It passes the template object, the module configuration, and the module object as arguments and expects no return value. It can be used e.g. to add additional data to the template.
+> **tag** It is available from version 3.5.0.
+
+
+## Parameters ##
+
+1. *FrontendTemplate* `$objTemplate`
+
+    The current frontend template instance for the article module
```
Comment (lionel-m) [inline api/hooks/compileArticle.md]: frontend -> front end

File: api/hooks/compileArticle.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# compileArticle
+
+The `compileArticle` hook is triggered after the article module has been compiled. It passes the template object, the module configuration, and the module object as arguments and expects no return value. It can be used e.g. to add additional data to the template.
```
Comment (lionel-m) [inline api/hooks/compileArticle.md]: expects no return value -> does not expect a return value

File: api/hooks/compileArticle.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# compileArticle
+
+The `compileArticle` hook is triggered after the article module has been compiled. It passes the template object, the module configuration, and the module object as arguments and does not expect a return value. It can be used e.g. to add additional data to the template.
+
+> **Tag** It is available from version 3.5.0.
```
Comment (aschempp) [inline api/hooks/compileArticle.md]: Please just write _Available from …_. Sorry about that, could have noticed that earlier ;-)

File: api/hooks/compileArticle.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# compileArticle
+
+The `compileArticle` hook is triggered after the article module has been compiled. It passes the template object, the module configuration, and the module object as arguments and does not expect a return value. It can be used e.g. to add additional data to the template.
```
Comment (aschempp) [inline api/hooks/compileArticle.md]: Please add line breaks after about 80 characters. Also, I think there shouldn't be a comma before _and_ ?

Comment (fiedsch) [inline api/hooks/compileArticle.md]: Not so sure about the comma. See for example http://www.grammarly.com/handbook/punctuation/comma/22/comma-before-and/

File: api/hooks/compileArticle.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+}
+```
+
+
+## More Information
+
+
+### Reference
+
+- [system/modules/ModuleArticle.php](https://github.com/contao/core/blob/master/system/modules/core/modules/ModuleArticle.php#L267)
```
Comment (aschempp) [inline api/hooks/compileArticle.md]: Please always link to a specific version, in this case it should be https://github.com/contao/core/blob/3.5.0/system/modules/core/modules/ModuleArticle.php#L267-L274

**Comment by aschempp** [comment]
Only one small issue left ;-)

**Comment by bezin** [comment]
Same here, I am sorry I forgot about those.

**Pull request merged**